python/mxnet/foo/nn/layer.py

```diff
@@ -207,6 +222,80 @@ def generic_forward(self, F, x, *args, **kwargs):
         raise NotImplementedError
 
 
+class _Binary(Layer):
```

Comment:
I still think this is confusing. Especially when it's overloading + and -.
How about we do it by inheriting layer for now and see if there is a need for this later?